src/extension/js/add-polyratings.js

```diff
@@ -205,6 +189,34 @@ export default class PolyratingIntegrator {
     })
   }
 
+  checkCalPolyRatings(rawname, nameElems) {
+    const [lastName, firstName] = rawname.split(/[, ]/)
+      .filter(name => name.length > 2 && name.split('I').length !== name.length + 1) // detects II and M.
```

Comment:
whats the purpose of detecting "II and M"?

Reply:
II is the titles like the first second and third. I also saw a teacher with m. before there name. I am doing this try and get more accurate first and last names since I don't iterate through names like your solution.